### ISSUE TITLE
Generated client that calls multiple servers (with different URLs)

### ISSUE THREAD

**Issue by gmicsa** [phoenixnap/springmvc-raml-plugin#129]
Hi,

Currently the generated client makes use of a base URL that is bound to a property and this way it`s not possible at runtime to call multiple servers that have different URLs. 
A possible solution would be to overload the generated client methods to contain also a baseUrl as a parameter.

Example for the sample project: 
Have 2 methods generated in the interface:

`    public ResponseEntity<List<GetDrinksResponse>> getDrinks();
    public ResponseEntity<List<GetDrinksResponse>> getDrinks(String baseUrl);
`

In the implementation, first method will delegate to the second method with baseUrl injected from the property.

What do you think? Could you implement this or do you see another possible solution to this use case?

Thanks,
Georgian

**Comment by kurtparis** [comment]
I personally favor using service discovery / load balancing  approaches like netflix ribbon/  f5 Vips etc. 

What you are saying can be done, and we could have it as a configuration option which is off by default. Are there any other users that might need this?

**Comment by gmicsa** [comment]
Thanks for the quick answer. 
Service discovery / load balancing is not an option for us as we are consuming an API implemented by different service providers (exposed on different URLs) and we need to decide at runtime to which service provider to dispatch the request.

**Comment by kurtparis** [comment]
Agreed, however as done by ribbon you can implement that logic that chooses a specific provider inside your httpclient or Rest template (by extending the class)  and the baseurl would only serve to identify the service.
